File: src/main/java/com/wgzhao/addax/admin/model/EtlColumn.java
package com.wgzhao.addax.admin.model;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.sql.Timestamp;

@Entity
@Table(name = "etl_column")
@Data
@IdClass(EtlColumnPk.class)
public class EtlColumn {

    @Id
    @Column(name = "tid", nullable = false)
    private Long tid;

    @Id
    @Column(name = "column_name", length = 255)
    private String columnName;

    @Column(name = "column_id")
    private int columnId;

    @Column(name = "source_type", length = 64)
    private String sourceType;

    @Column(name = "data_length")
    private int dataLength;

    @Column(name = "data_precision")
    private Integer dataPrecision;

    @Column(name = "data_scale")
    private Integer dataScale;

    @Column(name = "col_comment", length = 4000)
    private String colComment;

    @Column(name = "target_type", length = 50, nullable = false)
    private String targetType;

    @Column(name = "target_type_full", length = 100)
    private String targetTypeFull;

    @Column(name = "update_at")
    private Timestamp updateAt;
}

File: src/main/kotlin/com/wgzhao/addax/admin/model/EtlColumn.kt
package com.wgzhao.addax.admin.model

import jakarta.persistence.*
import java.sql.Timestamp

/**
 * 采集表字段信息实体类。
 * 用于描述采集表的字段结构、类型、注释等元数据信息。
 */
@Entity
@Table(name = "etl_column")
@IdClass(EtlColumnPk::class)
data class EtlColumn(
    @Id
    @Column(name = "tid", nullable = false)
    var tid: Long,

    @Column(name = "column_name", length = 255)
    var columnName: String,

    @Id
    @Column(name = "column_id")
    var columnId: Int = 0,

    @Column(name = "source_type", length = 64)
    var sourceType: String,

    @Column(name = "data_length")
    var dataLength: Int = 0,

    @Column(name = "data_precision")
    var dataPrecision: Int? = null,

    @Column(name = "data_scale")
    var dataScale: Int? = null,

    @Column(name = "col_comment", length = 4000)
    var colComment: String? = null,

    @Column(name = "target_type", length = 50, nullable = false)
    var targetType: String,

    @Column(name = "target_type_full", length = 100)
    var targetTypeFull: String,

    @Column(name = "update_at")
    var updateAt: Timestamp
)
